src/alert/__tests__/persistence-unit.test.tsx

```diff
@@ -0,0 +1,149 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { render, waitFor } from '@testing-library/react';
+
+import Alert, { AlertProps } from '../../../lib/components/alert';
+import { persistAlertDismiss, retrieveAlertDismiss } from '../../../lib/components/internal/persistence';
+import createWrapper from '../../../lib/components/test-utils/dom';
+
+jest.mock('../../../lib/components/internal/persistence', () => ({
+  persistAlertDismiss: jest.fn(),
+  retrieveAlertDismiss: jest.fn(),
+}));
+
+const mockPersistAlertDismiss = jest.mocked(persistAlertDismiss);
+const mockRetrieveAlertDismiss = jest.mocked(retrieveAlertDismiss);
+
+function renderAlert(props: AlertProps = {}) {
+  const { container } = render(<Alert {...props} />);
+  return {
+    getWrapper: () => createWrapper(container).findAlert(),
+    container,
+  };
+}
+
+describe('Alert Persistence', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockRetrieveAlertDismiss.mockResolvedValue(false);
+  });
+
+  describe('initialization', () => {
+    it('calls retrieveAlertDismiss when persistenceConfig is provided', async () => {
+      const persistenceConfig = { uniqueKey: 'test-alert' };
+      renderAlert({ persistenceConfig, dismissible: true });
+
+      await waitFor(() => {
+        expect(mockRetrieveAlertDismiss).toHaveBeenCalledWith(persistenceConfig);
```

Comment:
It is better to explicitly pass arguments instead of reusing the object. It will be easier to follow what data is being sent in and out

```suggestion
        expect(mockRetrieveAlertDismiss).toHaveBeenCalledWith({ uniqueKey: 'test-alert' });
```